src/main/java/com/entropyteam/entropay/employees/services/ContractService.java

```diff
@@ -28,14 +34,46 @@ public class ContractService extends BaseService<Contract, ContractDto, UUID> {
 
     @Autowired
     public ContractService(ContractRepository contractRepository, CompanyRepository companyRepository,
-            EmployeeRepository employeeRepository, RoleRepository roleRepository,
-            SeniorityRepository seniorityRepository) {
+                           EmployeeRepository employeeRepository, RoleRepository roleRepository,
+                           SeniorityRepository seniorityRepository) {
         this.contractRepository = contractRepository;
         this.companyRepository = companyRepository;
         this.employeeRepository = employeeRepository;
         this.roleRepository = roleRepository;
         this.seniorityRepository = seniorityRepository;
+    }
+
+    @Transactional
+    @Override
+    public ContractDto create(ContractDto contractDto) {
+        contractRepository.findContractByEmployeeIdAndActiveIsTrue(contractDto.employeeId())
+                .ifPresent(existent -> {
+                    existent.setActive(false);
+                    existent.setEndDate(LocalDate.now());
+                    contractRepository.saveAndFlush(existent);
+                });
+        return super.create(contractDto.withActive(true));
+    }
+
+    @Transactional
+    public ContractDto modifyStatus(UUID contractId, boolean setActive) {
+        Contract contract = contractRepository.findById(contractId)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Entity not found"));
+
+        if ((!setActive && !contract.isActive()) || (setActive && contract.isActive()))
+            throw new ResponseStatusException(HttpStatus.NO_CONTENT, "Contract " + contract.getId() + " already " + (setActive ? "active" : "inactive")
+                    + " for employee " + contract.getEmployee().getId());
 
+        if (setActive) {
+            contractRepository.findContractByEmployeeIdAndActiveIsTrue(contract.getEmployee().getId()).ifPresent(existent -> {
+                existent.setActive(false);
+                existent.setEndDate(LocalDate.now());
```

Comment:
please, also update the modified at: `existent.setModifiedAt(LocalDateTime.now());`